algorithms/scripts/LinearSVC-htcai.py

```diff
@@ -0,0 +1,323 @@
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel('False Positive Rate')
+plt.ylabel('True Positive Rate')
+plt.title('Predicting TP53 mutation from gene expression (ROC curves)')
+plt.legend(loc='lower right');
+
+
+# ## What are the classifier coefficients?
+
+# In[24]:
+
+coef_df = pd.DataFrame(best_clf.coef_.transpose(), index=X.columns[feature_mask], columns=['weight'])
+coef_df['abs'] = coef_df['weight'].abs()
+coef_df = coef_df.sort_values('abs', ascending=False)
+
+
+# In[25]:
+
+'{:.1%} zero coefficients; {:,} negative and {:,} positive coefficients'.format(
+    (coef_df.weight == 0).mean(),
+    (coef_df.weight < 0).sum(),
+    (coef_df.weight > 0).sum()
+)
+
+
+# In[26]:
+
+coef_df.head(10)
+
+
+# The results are not surprising. TP53 is a transcription modulator and when it mutated in a tumor, the cell goes haywire. This makes finding a transcriptional signature fairly easy. Also, the genes that the classifier uses is interesting, but not necessarily novel.
+# 
+# 1. TP53 is a [transcription factor](https://en.wikipedia.org/wiki/Transcription_factor "TF wiki") that regulates many genes including EDA2R. Studies have linked EDA2R (or XEDAR) to [increased survival in colon cancer patients](http://www.ncbi.nlm.nih.gov/pubmed/19543321) and [losing hair as a result of chemotherapy](http://onlinelibrary.wiley.com/doi/10.1016/j.febslet.2010.04.058/full)
+# 2. SPATA18 is a gene associated with spermatogenesis and is a transcription factor for TP53. It's association with TP53 was [recently discovered](http://www.ncbi.nlm.nih.gov/pubmed/21300779) in 2011.
+# 3. C6orf138 (or [PTCHD4](http://www.genecards.org/cgi-bin/carddisp.pl?gene=PTCHD4)) is also a transcriptional target for TP53 and was only recently discovered in [2014 to repress hedgehog signalling](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4239647/).
+# 4. The list goes on and includes several other TP53 targets...
+
+# ## Investigate the predictions
+
+# In[27]:
+
+predict_df = pd.DataFrame.from_items([
+    ('sample_id', X.index),
+    ('testing', X.index.isin(X_test.index).astype(int)),
+    ('status', y),
+    ('decision_function', pipeline.decision_function(X)),
+    ('probability', pipeline.predict(X)),
+])
+predict_df['probability_str'] = predict_df['probability'].apply('{:.1%}'.format)
+
+
+# In[28]:
+
+# Top predictions amongst negatives (potential hidden responders)
+predict_df.sort_values('decision_function', ascending=False).query("status == 0").head(10)
+
+
+# In[29]:
+
+# Ignore numpy warning caused by seaborn
+warnings.filterwarnings('ignore', 'using a non-integer number instead of an integer')
+
+ax = sns.distplot(predict_df.query("status == 0").decision_function, hist=False, label='Negatives', color = 'red')
+ax = sns.distplot(predict_df.query("status == 1").decision_function, hist=False, label='Positives', color = 'green')
+
+
+# In[30]:
+
+from sklearn.metrics import confusion_matrix
+from tabulate import tabulate
```

Comment:
Can you use pandas instead to make a dataframe from `conf_mat`? In short, I don't think we should create a `tabulate` dependency for a trivial task where pandas already provides the functionality.